v2/internal/controller/controllers/operatormode_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/microsoft.resources/v1alpha1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/controller/config"
+	"github.com/Azure/azure-service-operator/v2/internal/controller/testcommon"
+)
+
+func TestOperatorMode_Webhooks(t *testing.T) {
+	t.Parallel()
+	tc := globalTestContext.ForTestWithConfig(t, config.Values{
+		OperatorMode: config.OperatorModeWebhooks,
+	})
+
+	rg := resources.ResourceGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tc.Namer.GenerateName("rg"),
+			Namespace: tc.Namespace(),
+		},
+		Spec: resources.ResourceGroupSpec{
+			Location: tc.AzureRegion,
+			Tags:     testcommon.CreateTestResourceGroupDefaultTags(),
+		},
+	}
+	tc.Expect(rg.Spec.AzureName).To(Equal(""))
+
+	_, err := tc.CreateResourceGroup(&rg)
+	tc.Expect(err).ToNot(HaveOccurred())
+	// AzureName should have been defaulted on the group on the
+	// way in (it doesn't require waiting for a reconcile).
+	// TODO: this works here, but it's possibly because envtest
```

Comment:
Not 100% sure what you are getting at with this TODO, but I think that this assert would work even if APIServer was remote. Why do you think it wouldn't?

Reply:
As discussed on teams, delete this I think since it is guaranteed